src/renderer/components/Login.js

```diff
@@ -51,7 +83,18 @@ class Login extends React.Component {
       send_server: this.state.sendServer,
       send_user: this.state.sendUser,
       send_pw: this.state.sendPw,
-      send_port: this.state.sendPort
+      send_port: this.state.sendPort,
+      server_flags: this.translateSecurityToServerFlags(
+        this.state.mailSecurity,
+        this.state.sendSecurity
+      )
+    }
+
+    // Remove all properties where value is an empty string, gets us around
+    // React not liking state properties where the value is null/undefined if
+    // bound to an input value
+    for (let key in credentials) {
+      if (credentials[key] === '') delete credentials[key]
     }
```

Comment:
when would it be null/undefined?

Reply:
React doesn't like having a null or undefined value bound to an input value, i get a warning and it should be an empty string. But empty strings aren't seen as nothing in the core/login procedure so i'm cleaning them out. Otherwise we're doing a set_config for all keys.